### PR TITLE
[stable9] Add support to know where the storage test comes from

### PR DESCRIPTION
**Pull request by jvillafanez**
(manual) backport of https://github.com/owncloud/core/pull/25008 to stable9 . Minor changes have been added (mainly boolean conversions that weren't required for master)

Required for https://github.com/owncloud/windows_network_drive/pull/408

### CODE REVIEW

File: apps/files_external/controller/globalstoragescontroller.php [hunk 1]
```diff
@@ -143,8 +143,10 @@ public function update(
 		$mountOptions,
 		$applicableUsers,
 		$applicableGroups,
-		$priority
+		$priority,
+		$testOnly = true
 	) {
+		$testOnly = filter_var($testOnly, FILTER_VALIDATE_BOOLEAN);  // boolean conversion required
```
Comment (DeepDiver1975) [inline apps/files_external/controller/globalstoragescontroller.php]: as discussed in #25173 - use PHPDoc

**Comment by DeepDiver1975** [comment]
@jvillafanez please squash - thx

**Comment by DeepDiver1975** [comment]
took care about the squash

**Comment by icewind1991** [comment]
Code looks good :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**